Formula/aerc.rb

```diff
@@ -16,9 +17,12 @@ class Aerc < Formula
   end
 
   depends_on "go" => :build
+  depends_on "make" => :build
```

Comment:
Why didn't previous CI builds fail without this?

Reply:
The only things using != aren't build critical.

The most user-visible one is running `aerc -v`, which will output just `aerc`, without any version.

Reply:
Can we improve the test to catch that kind of errors in the future?

Reply:
I'm not really sure how other than hardcoding the version number. Maybe parsing a regex for the _form_ of the version number?

Reply:
Is the version the same as the version that `brew` thinks this is? (See `brew info aerc`.)

If so, then you can use `version` instead of hard-coding the number.

Reply:
No, the version is something like `aerc 0.11.0-125-g8eab82964986 (go1.19 arm64 darwin)` (for a "random" git commit build not compiled with Homebrew).

Reply:
Oh, I see. Can we just invoke `go build` directly instead of the extra indirection through `make`?

Reply:
Not really, there's quite a lot of extra logic in the Makefile, as well as the manpage processing.

Reply:
How much extra? If we can do it in ~10 lines of Ruby code (and one can do a lot with that), then we should.

Reply:
Here's what the Makefile does:

1. The version is [set](https://git.sr.ht/~rjarry/aerc/tree/master/item/Makefile#L5) with `git-describe`.
2. `GOFLAGS` are base64 encoded and passed as the `main.Flags` variable. This won't be needed since no flags are passed.
3. `config.shareDir` (used for getting config/stylesets) is [set](https://git.sr.ht/~rjarry/aerc/tree/master/item/Makefile#L19) based on `$(SHAREDIR)`, which is set to `$(PREFIX)/share/aerc`.
4. Go build is [executed](https://git.sr.ht/~rjarry/aerc/tree/master/item/Makefile#L40)
5. Each scdoc file is compiled (L80-L87)
6. Everything is installed (9 directories created, 27 files installed) (L97-L127)

Reply:
Yea, seems like you can get that done with a few lines of Ruby.

Reply:
@rjarry What do you think?

Reply:
I feel like using ruby would bloat the process… a few lines of shell script would do just fine.

```sh
#!/bin/sh
make "VERSION=$(git describe --long --tags --dirty 2>/dev/null)"
```

Not very into the entire Homebrew ecosystem, but this seems more pragmatic.

Reply:
This will probably be translated into Ruby at some point in order to standardise handling of LDFLAGS, so skipping the `Makefile` will likely happen at some point.

Though, if it suffices to pass a `VERSION` argument to `make`, then one doesn't even need a shell script to do that.

Not to mention that `git describe` will not work because the working directory is not a Git repo.

Reply:
rjarry said in the #aerc IRC channel that passing `VERSION` to `make` is the correct thing to do. Should the revision bump still be present if that's the only change?

Reply:
I'd leave the `revision` bump out. We can bump it if enough users complain about malformed `--version` output from existing installs, but that's pretty unlikely.

I'd strongly suggest just calling `go build` and `scdoc` directly, though. This formula will do that eventually anyway.

Reply:
What amount of customizability is lacking with the Makefile? Wouldn't it be better to get customizable options upstream and then use a supported build method?

`LDFLAGS` can already be changed with [`GO_EXTRA_LDFLAGS`](https://git.sr.ht/~rjarry/aerc/tree/master/item/Makefile#L20), I think.

Reply:
Ok, then can we parse the content of `std_go_args` and check that the flags (other than `-o`) are passed to `go build`? Not all flags will be `-ldflags`.

Reply:
There's [`GOFLAGS`](https://git.sr.ht/~rjarry/aerc/tree/master/item/Makefile#L12).

Reply:
@carlocab switched to just setting `VERSION` in https://github.com/Homebrew/homebrew-core/commit/3f63073a287a63307279cceb1d654c52b88fcb7a

Reply:
Should I check if the `aerc -v` output in the test contains `#{version}`?

Reply:
> Ok, then can we parse the content of `std_go_args` and check that the flags (other than `-o`) are passed to `go build`? Not all flags will be `-ldflags`.

To clarify: this was a suggestion for your PR, since you seem reluctant to call `go build` directly.

Reply:
Oh, sorry.

When echoing the `std_go_args` for aerc, it's just `-trimpath` and `-o=/opt/homebrew/Cellar/aerc/0.11.0/bin/aerc`, which are already set by the Makefile and `PREFIX`.

Is this just so when the args are changed in future it'll be added here?

Reply:
Will ldflags get added to `std_go_args`? If so, I'd need to split it by the `-ldflags` argument and pass it to `GOFLAGS`/`GO_EXTRA_LDFLAGS`, right?